src/RadForce/test_radiation_force.cpp

```diff
@@ -66,14 +59,42 @@ template <> struct Physics_Traits<TubeProblem> {
 	static constexpr bool is_radiation_enabled = true;
 	// face-centred
 	static constexpr bool is_mhd_enabled = false;
+	// number of radiation groups
+	static constexpr int nGroups = 2;
 };
 
-template <> AMREX_GPU_HOST_DEVICE auto RadSystem<TubeProblem>::ComputePlanckOpacity(const double /*rho*/, const double /*Tgas*/) -> double
+template <> struct RadSystem_Traits<TubeProblem> {
+	static constexpr double c_light = c_light_cgs_;
+	static constexpr double c_hat = 10. * (Mach1 * a0);
+	static constexpr double radiation_constant = radiation_constant_cgs_;
+	static constexpr double Erad_floor = 0.;
+	static constexpr bool compute_v_over_c_terms = true;
+	static constexpr double energy_unit = C::ev2erg;
+	static constexpr amrex::GpuArray<double, Physics_Traits<TubeProblem>::nGroups + 1> radBoundaries{0., 13.6, inf}; // eV
+};
+
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<TubeProblem>::ComputePlanckOpacity(const double /*rho*/, const double /*Tgas*/)
+    -> quokka::valarray<double, Physics_Traits<TubeProblem>::nGroups>
 {
-	return 0.; // no heating/cooling
+	quokka::valarray<double, Physics_Traits<TubeProblem>::nGroups> kappaPVec{};
+	// kappaPVec.fillin(0.);
+	for (int g = 0; g < nGroups_; ++g) {
+		kappaPVec[g] = 0.; // no heating/cooling
+	}
+	return kappaPVec;
 }
 
-template <> AMREX_GPU_HOST_DEVICE auto RadSystem<TubeProblem>::ComputeRosselandOpacity(const double /*rho*/, const double /*Tgas*/) -> double { return kappa0; }
+template <>
+AMREX_GPU_HOST_DEVICE auto RadSystem<TubeProblem>::ComputeFluxMeanOpacity(const double /*rho*/, const double /*Tgas*/)
+    -> quokka::valarray<double, Physics_Traits<TubeProblem>::nGroups>
+{
+	quokka::valarray<double, Physics_Traits<TubeProblem>::nGroups> kappaFVec{};
+	// kappaFVec.fillin(kappa0);
```

Comment:
Can you remove commented-out code? Unless there's a very good reason to keep it, it just makes it less readable to future developers/users. We've tried to avoid keeping commented-out code anywhere in Quokka (although there are a few places where it still needs to be removed).

Reply:
I removed it and all other similar things.